src/features/navigator/NarrativeControl/Share.tsx

```diff
@@ -223,44 +341,74 @@ export const Share: FC<{
       }
     }
   }, [dispatch, wsPermsQuery.currentData, wsPermsQuery.isSuccess]);
-  const shareSelectId = useId();
+
+  /* derived values */
   const narrativeIsPublic = usernameShares['*'] === 'r';
   const userPermission: UserPermission =
     (usernameAuthed && usernameShares[usernameAuthed]) || 'n';
-  const userIsAdmin = usernameAuthed && userPermission === 'a';
+  const userIsAdmin = Boolean(usernameAuthed) && userPermission === 'a';
   const usernames = usernameAuthed ? [usernameAuthed] : [];
+  const usernamePermissions = filterUsernames({
+    usernames,
+    results: usernameShares,
+  });
+
+  /* add permission callback */
+  // See <SelectUser /> component
+
+  /* set permission callback */
+  // See <UserPermissionControl /> component
+
+  /* remove permission callback */
+  // See <UserPermissionControl /> component
```

Comment:
I'm not sure I get why these comments are here as things like `setUserPermission` are already listed as imports

Reply:
I have been organizing these control menu components like this:
```js
/* hooks */
/* derived values */
/* callback */
/* render */
```

except Share has 4 different callbacks and it made the most sense to me to put them in these other components, so it's just a reminder (mainly for me) to know where to look for the callback corresponding to the listed behavior. I was also thinking that this module is getting a little long and could probably serve to be split, but never actually did that since it ended up less than 500 lines for now.